src/pkgcheck/checks/codingstyle.py

```diff
@@ -1539,3 +1539,58 @@ def feed(self, pkg: bash.ParseTree):
             if len(nodes) > 1:
                 lines = sorted(node.start_point[0] + 1 for node in nodes)
                 yield DuplicateFunctionDefinition(func_name, lines=lines, pkg=pkg)
+
+
+class VariableOrderWrong(results.VersionResult, results.Style):
+    """Variable were defined in an unexpected error."""
+
+    def __init__(self, first_var, second_var, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.first_var = first_var
+        self.second_var = second_var
+
+    @property
+    def desc(self):
+        return f"variable {self.first_var} should occur before {self.second_var}"
+
+
+class VariableOrderCheck(Check):
+    """Scan ebuilds for variables defined in a different order than skel.ebuild dictates."""
+
+    _source = sources.EbuildParseRepoSource
+    known_results = frozenset({VariableOrderWrong})
+
+    # Order from skel.ebuild
+    variable_order = [
+        "EAPI",
+        "DESCRIPTION",
+        "HOMEPAGE",
+        "SRC_URI",
+        "S",
+        "LICENSE",
+        "SLOT",
+        "KEYWORDS",
+        "IUSE",
+        "RESTRICT",
+        "RDEPEND",
+        "DEPEND",
+        "BDEPEND",
+        "IDEPEND", # Temporarily missing from skel.ebuild
+    ]
+
+    def feed(self, pkg: bash.ParseTree):
+        var_assigns = []
+
+        for node in pkg.tree.root_node.children:
+            if node.type == "variable_assignment":
+                used_name = pkg.node_str(node.child_by_field_name("name"))
+                var_assigns.append(used_name)
```

Comment:
let's include the variables only if intersting:

```suggestion
                if used_name in self.variable_order:
                    var_assigns.append(used_name)
```